### PR TITLE
Feature/filters on the map

### PR DESCRIPTION
**Pull request by ykobuk**
Add filters by category and days

### CODE REVIEW

File: client/src/components/MapLayout/index.jsx [hunk 1]
```diff
@@ -1,35 +1,75 @@
...
+          category: this.state.category
+      }
+      request('/events-short-info', 'POST', JSON.stringify(filterData)).then(data => {
+          if ('info' in data) {
+              this.setState({events: []})
+              data.info.forEach(element => {
+                  let eventShortInfo = this.state.events;
+                  eventShortInfo.push(element);
+                  this.setState({ events: eventShortInfo });
+                  console.log()
```
Comment (Merteg) [inline client/src/components/MapLayout/index.jsx]: I think you can replace strings 27-32 by `this.setState({ events: data.info })`

File: client/src/components/MapLayout/index.jsx [hunk 2]
```diff
@@ -1,35 +1,75 @@
...
-    request('/events-short-info').then(data => {
-      if ('info' in data) {
-        data.info.forEach(element => {
-          let eventShortInfo = this.state.events;
-          eventShortInfo.push(element);
-          this.setState({ events: eventShortInfo });
-        });
+      let filterData = {
+          day_filter: this.state.day_filter
+
```
Comment (Merteg) [inline client/src/components/MapLayout/index.jsx]: Useless blank line :)

File: client/src/components/MapLayout/index.jsx [hunk 3]
```diff
@@ -1,35 +1,75 @@
...
       }
-    });
+      request('/events-short-info', 'POST', JSON.stringify(filterData)).then(data => {
+          if ('info' in data) {
+              data.info.forEach(element => {
+                  let eventShortInfo = this.state.events;
+                  eventShortInfo.push(element);
+                  this.setState({ events: eventShortInfo });
+
```
Comment (Merteg) [inline client/src/components/MapLayout/index.jsx]: The same as on lines 27-32

File: client/src/components/MapLayout/index.jsx [hunk 4]
```diff
@@ -1,35 +1,75 @@
...
+              });
+          }
+      });
   }
 
   render(){
     return (
       <Row>
+          <Input id='day_filter' s={3} type='select' label="Filter by days" defaultValue='' onChange={this.onChangeHandler}>
```
Comment (Merteg) [inline client/src/components/MapLayout/index.jsx:46]: You have 14 days by default in the state, so I think it's better to set it by default here.

File: server/server/views/location_event.py [hunk 5]
```diff
@@ -1,18 +1,33 @@
...
                             cors_origins=('http://localhost:3000',))
 
 
-@events_short_info.get()
+@events_short_info.post()
 def get_events_short_info(request):
     """ """
-    responce = {
+    datetime_now = datetime.now()
+    categories = {'holiday': 1, 'concert': 2, 'movie': 3, 'Voluntary': 4}
```
Comment (Merteg) [inline server/server/views/location_event.py]: Hardcoding is bad, we have get_all() method in the category model, you can use it.

File: server/server/views/location_event.py [hunk 6]
```diff
@@ -1,18 +1,33 @@
...
-    responce['info'] = Event.get_events_short_info(request, 14)
-    return responce
+    if json['category'] in categories:
+        json['category'] = categories[json['category']]
+
+    if json['category'] == '':
+        response['info'] = Event.get_events_short_info(request, int(json['day_filter']))
+        return response
+    else:
+        response['info'] = request.dbsession.query(Event).\
```
Comment (Merteg) [inline server/server/views/location_event.py]: Connections to the db must be in models

**Review by Merteg — CHANGES_REQUESTED** [review]

File: client/src/components/MapLayout/index.jsx (hunk 2 above)
Comment (ykobuk) [reply client/src/components/MapLayout/index.jsx]: I've left it specially for you =)

**Comment by ykobuk** [comment]
Corrected the problems that Merteg pointed out.

File: server/server/views/location_event.py [hunk 7]
```diff
@@ -11,23 +11,22 @@
...
+    categories_list = [obj.category for obj in categories]
+    categories_id = [obj.id for obj in categories]
+    categories_dict = dict(zip(categories_list, categories_id))
     response = {
         'info': []
     }
-    if json['category'] in categories:
-        json['category'] = categories[json['category']]
+    if json['category'] in categories_list:
+        json['category'] = categories_dict[json['category']]
```
Comment (Merteg) [inline server/server/views/location_event.py]: I thought a bit, you get a category string from a client, and you need to get it id. So you don't need to get all categories, just get one. (There are a wonderful method in Category model - `get_by_name`)

File: client/src/components/MapLayout/index.jsx [hunk 8]
```diff
@@ -9,7 +9,7 @@ import SelectCategory from '../EventForm/SelectCategory';
 class MapLayout extends Component {
   state = {
     events: [],
-    day_filter: 14,
+    day_filter: 0,
```
Comment (Merteg) [inline client/src/components/MapLayout/index.jsx]: Why did you remove that? When map renders for the first time, it's don't have any markers now.

**Review by Merteg — CHANGES_REQUESTED** [review]

File: client/src/components/MapLayout/index.jsx (hunk 8 above)
Comment (ykobuk) [reply client/src/components/MapLayout/index.jsx]: Because user won't know what is it

Comment (Merteg) [reply client/src/components/MapLayout/index.jsx]: Hm, I see we must to discuss it

**Comment by ykobuk** [comment]
refactored the location_event.py

**Review by Merteg — APPROVED** [review]
